Sources/sourcekit-lsp/main.swift

```diff
@@ -18,134 +20,177 @@ import LSPLogging
-  }
-  if let flags = parsedArguments.get(buildFlagsCxx) {
-    result.serverOptions.buildSetup.flags.cxxCompilerFlags = flags
-  }
-  if let flags = parsedArguments.get(buildFlagsLinker) {
-    result.serverOptions.buildSetup.flags.linkerFlags = flags
-  }
-  if let flags = parsedArguments.get(buildFlagsSwift) {
-    result.serverOptions.buildSetup.flags.swiftCompilerFlags = flags
-  }
+extension LogLevel: ExpressibleByArgument {}
+extension BuildConfiguration: ExpressibleByArgument {}
 
-  if let options = parsedArguments.get(clangdOptions) {
-    result.serverOptions.clangdOptions = options
-  }
+struct Main: ParsableCommand {
+  static let configuration = CommandConfiguration(
+    abstract: "Language Server Protocol implementation for Swift and C-based languages"
+  )
 
-  if let path = parsedArguments.get(indexStorePath)?.path {
-    result.serverOptions.indexOptions.indexStorePath = path
-  }
-  if let path = parsedArguments.get(indexDatabasePath)?.path {
-    result.serverOptions.indexOptions.indexDatabasePath = path
-  }
-
-  if let logLevel = parsedArguments.get(loggingOption) {
-    Logger.shared.currentLevel = logLevel
-  } else {
-    Logger.shared.setLogLevel(environmentVariable: "SOURCEKIT_LOGGING")
-  }
-
-  if let sync = parsedArguments.get(syncOption) {
-    result.syncRequests = sync
+  /// Whether to wait for a response before handling the next request.
+  /// Used for testing.
+  @Flag(name: .customLong("sync"))
+  var syncRequests = false
+
+  @Option(help: "Set the logging level (debug|info|warning|error)")
+  var logLevel: LogLevel = LogLevel.default
+
+  @Option(
+    name: [.customLong("configuration"), .customShort("c")],
+    help: "Build with configuration (debug|release)"
+  )
+  var buildConfiguration = BuildConfiguration.debug
+
+  @Option(help: "Specify build/cache directory")
+  var buildPath: AbsolutePath?
+
+  @Option(
+    name: .customLong("Xcc", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all C compiler invocations"
+  )
+  var buildFlagsCc = [String]()
+
+  @Option(
+    name: .customLong("Xcxx", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all C++ compiler invocations"
+  )
+  var buildFlagsCxx = [String]()
+
+  @Option(
+    name: .customLong("Xlinker", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all linker invocations"
+  )
+  var buildFlagsLinker = [String]()
+
+  @Option(
+    name: .customLong("Xswiftc", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all Swift compiler invocations"
+  )
+  var buildFlagsSwift = [String]()
+
+  @Option(
+    name: .customLong("Xclangd", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass options to clangd command-line"
+  )
+  var clangdOptions = [String]()
+
+  @Option(
+    name: .customLong("index-store-path", withSingleDash: true),
```

Comment:
I suggest accepting both single- and double-dash for this option if possible

Reply:
@natecook1000 please correct me if I'm wrong, but I don't think that accepting both is currently possible.

Reply:
🤔 It looks like this is prohibited by the command validation, but doesn't necessarily need to be. Opened an issue to allow it: https://github.com/apple/swift-argument-parser/issues/231

Reply:
@MaxDesiatov This fix is included in version 0.3.1! 👍

Reply:
This is great, thank you!